ktor-io/common/src/io/ktor/utils/io/ByteChannelSequential.kt

```diff
@@ -871,8 +880,9 @@ public abstract class ByteChannelSequentialBase(
             val buffer = request(1) ?: IoBuffer.Empty
             if (buffer.readRemaining > offset) {
                 buffer.discardExact(offset)
-                bytesCopied = minOf(buffer.readRemaining.toLong(), max)
+                bytesCopied = minOf(buffer.readRemaining.toLong(), max, destination.size - destinationOffset)
                 buffer.memory.copyTo(destination, buffer.readPosition.toLong(), bytesCopied, destinationOffset)
+                buffer.discardExact(bytesCopied)
```

Comment:
Doesn't it lead to consuming bytes instead of peeking into?

Reply:
thanks, updated